Write repmgr conf and pgpass when configuring a primary database server

carbonin opened the pull request:
When a primary server fails and a failover is executed it is typical to reintroduce the failed primary server as a standby if the failure is recoverable.

Some of the configuration edits made when configuring a standby also should be done when configuring a primary server to deal with this situation.

These config changes are safe to make when the server is running as a primary and will allow us to reintroduce the server as a standby without having to make the changes manually.

@gtanzillo @jvlcek please review

jvlcek commented on gems/pending/appliance_console/database_replication.rb:
```diff
@@ -70,6 +76,16 @@ def generate_cluster_name
+
+      FileUtils.chmod(0600, PGPASS_FILE)
+      FileUtils.chown("postgres", "postgres", PGPASS_FILE)
+    end
```
**Suggestion:**
Although this will work it might be best to return `true` here. It would avoid having a `string literal` in the condition at the caller.

jvlcek commented on gems/pending/spec/appliance_console/database_replication_primary_spec.rb:
```diff
@@ -66,29 +66,9 @@
-      expect(subject.activate).to be false
-    end
-
-    it "returns false when generate_cluster_name fails" do
```
Why remove this test? Doesn't it help with code coverage?

jvlcek commented on gems/pending/spec/appliance_console/database_replication_primary_spec.rb:
```diff
@@ -66,29 +66,9 @@
-      expect(subject.activate).to be false
-    end
-
-    it "returns false when initialize_primary_server fails" do
```
Same here, why remove this test? Doesn't it help with code coverage?

jvlcek commented on gems/pending/spec/appliance_console/database_replication_standby_spec.rb:
```diff
@@ -58,81 +58,29 @@
-      expect(subject.activate).to be false
-    end
-
-    it "returns false when start_postgres fails" do
```
@carbonin Please describe why you are removing these tests. Don't they help with code coverage?

carbonin commented:
@jvlcek regarding the specs I removed:

They don't change code coverage because they were the short circuited versions which are only testing that the `&&` operator works as expected. I left the version that runs through each call and tests the newly added conditional logic on `run_repmgrd_configuration`, which should be sufficient for testing syntax and covering the entire method.

They are also difficult to maintain as they need to be changed for any change in the `activate` method even if, for example, the behavior remained the same, but we split one of the methods being called.

jvlcek commented:
@carbonin Thank you for elaborating.

carbonin commented:
Closing in favor of https://github.com/ManageIQ/manageiq-gems-pending/pull/1 now that manageiq-gems-pending is extracted.
